packages/gatsby/src/schema/node-model.js

```diff
@@ -267,6 +267,42 @@ class LocalNodeModel {
 
     const nodeTypeNames = toNodeTypeNames(this.schema, gqlType)
 
+    let runQueryActivity
+
+    // check if we can get node by id and skip
+    // more expensive query pipeline
+    if (
+      typeof query?.filter?.id?.eq !== `undefined` &&
+      Object.keys(query.filter).length === 1 &&
+      Object.keys(query.filter.id).length === 1
+    ) {
+      if (tracer) {
+        runQueryActivity = reporter.phantomActivity(`runQuerySimpleIdEq`, {
```

Comment:
What's the goal of this phantomActivity? To see it in tracing?

Reply:
Yes, just for instrumentation. This code path skips `runQuery` activity, so I replaced it with something so traces make more sense